extensions/contract-termination/src/main/java/de/sovity/edc/extension/contacttermination/query/ContractAgreementTerminationDetailsQuery.java

```diff
@@ -0,0 +1,53 @@
+/*
+ *  Copyright (c) 2024 sovity GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       sovity GmbH - initial API and implementation
+ *
+ */
+
+package de.sovity.edc.extension.contacttermination.query;
+
+import de.sovity.edc.extension.contacttermination.ContractAgreementTerminationDetails;
+import lombok.RequiredArgsConstructor;
+import lombok.val;
+import org.eclipse.edc.connector.contract.spi.types.negotiation.ContractNegotiation;
+import org.jooq.DSLContext;
+
+import static de.sovity.edc.ext.db.jooq.Tables.EDC_CONTRACT_AGREEMENT;
+import static de.sovity.edc.ext.db.jooq.Tables.EDC_CONTRACT_NEGOTIATION;
+import static de.sovity.edc.ext.db.jooq.Tables.SOVITY_CONTRACT_TERMINATION;
+
+@RequiredArgsConstructor
+public class ContractAgreementTerminationDetailsQuery {
+
+    public ContractAgreementTerminationDetails fetchAgreementDetailsOrThrow(DSLContext dsl, String agreementId) {
+        val n = EDC_CONTRACT_NEGOTIATION;
+        val a = EDC_CONTRACT_AGREEMENT;
+        val t = SOVITY_CONTRACT_TERMINATION;
+
+        return dsl.transactionResult(trx ->
+            trx.dsl().select(
```

Comment:
- To prevent having to deal with this two-step API we should encorporate it into the DslContextFactory
- Mostly you don't want to have to go "trx.dsl()" all the time compared to "dsl" 
- Having a "dsl" compete against some imported "dslContext"
- This is just asking for human errors.